linera-core/src/join_set_ext.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! An extension trait to allow determining at compile time how tasks are spawned on the Tokio
+//! runtime.
+//!
+//! In most cases the [`Future`] task to be spawned should implement [`Send`], but that's
+//! not possible when compiling for `wasm32-unknown-unknown`. In that case, the task is
+//! spawned in a [`LocalSet`][`tokio::tast::LocalSet`].
+
+use std::future::Future;
+
+use tokio::{
+    sync::oneshot,
+    task::{AbortHandle, JoinSet},
+};
+
+/// An extension trait for the [`JoinSet`] type.
+#[cfg(not(target_arch = "wasm32"))]
+pub trait JoinSetExt: Sized {
+    /// Spawns a `future` task on this [`JoinSet`] using [`JoinSet::spawn`].
+    ///
+    /// Returns a [`oneshot::Receiver`] to receive the `future`'s output, and an
+    /// [`AbortHandle`] to cancel execution of the task.
```

Comment:
Would it be better to define the trait only once, and explain the feature flag in the comment here? It would make it more discoverable.